### PR TITLE
Test issue_generation on pull requests.

### PR DESCRIPTION
**Pull request by granluo**

### CODE REVIEW

**Comment by granluo** [comment]
Add a new testing job to avoid interfering issues labeled `nightly-testing`.

File: .github/actions/testing_report_generation/app.rb [hunk 1]
```diff
@@ -20,7 +20,9 @@
 require 'tzinfo'
 
 REPO_NAME_WITH_OWNER = ENV['GITHUB_REPOSITORY']
-NO_WORKFLOW_RUNNING_INFO = 'All nightly cron job were not run. Please make sure there at least exists one cron job running.'.freeze
+GITHUB_WORKFLOW_URL = "https://github.com/#{REPO_NAME_WITH_OWNER}/actions/runs/#{ENV['GITHUB_RUN_ID']}"
+TESTS_TIME_INTERVAL = 86400
+NO_WORKFLOW_RUNNING_INFO = "All nightly cron job were not run in the last #{TESTS_TIME_INTERVAL} secs. Please review [log](#{GITHUB_WORKFLOW_URL}) make sure there at least exists one cron job running.".freeze
```
Comment (paulb777) [inline .github/actions/testing_report_generation/app.rb:25]: Consider converting to hours

File: .github/workflows/generate_issues.yml [hunk 2]
```diff
@@ -21,3 +22,19 @@ jobs:
         access-token: '${{ secrets.GITHUB_TOKEN }}'
         # This is to exclude workflows that will be searched in the nightly report.
         exclude-workflow-files: 'performance-integration-tests.yml'
+
+  test_generate_an_issue:
+    # Don't run on private repo.
+    if: github.repository == 'Firebase/firebase-ios-sdk' && github.event_name != 'schedule'
```
Comment (paulb777) [inline .github/workflows/generate_issues.yml:28]: Is this only for testing? How does it run if it's testing `!= 'schedule'`?

Comment (granluo) [reply .github/workflows/generate_issues.yml:28]: The job will be triggered by either pull_request or schedule events.
This job will only be triggered by updating paths under pull_requests, which here are 
```
    - '.github/workflows/generate_issues.yml'
    - '.github/actions/testing_report_generation**'
```

Comment (paulb777) [reply .github/workflows/generate_issues.yml:28]: Then shouldn't it be `github.event_name == 'schedule'`?

Comment (granluo) [reply .github/workflows/generate_issues.yml:28]: It is testing the github action tool and generate a new issue. It is mimicking the scheduled job, so it is not testing `!= 'schedule'`

**Review by paulb777 — APPROVED** [review]

**Pull request merged**